packages/data-context/test/unit/sources/RelevantRunsDataSource.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { expect } from 'chai'
+import sinon from 'sinon'
+
+import { DataContext } from '../../../src'
+import { createTestDataContext } from '../helper'
+import { RelevantRunsDataSource, EMPTY_RETURN } from '../../../src/sources'
+
+const TEST_SHA = 'fcb90fc753a2111d1eb32d207e801e6b2985a231'
+
+describe('RelevantRunsDataSource', () => {
+  let ctx: DataContext
+  let dataSource: RelevantRunsDataSource
+
+  beforeEach(() => {
+    ctx = createTestDataContext('open')
+    dataSource = new RelevantRunsDataSource(ctx)
+    //sinon.stub(ctx.util, 'fetch').resolves({} as any)
+    // sinon.stub(ctx.util, 'fetch').callsFake(() => {
+    //   console.log(args)
+    // })
+  })
+
+  it('returns empty with no shas', async () => {
+    const result = await dataSource.getRelevantRuns([])
+
+    expect(result).to.equal(EMPTY_RETURN)
+  })
+
+  it('returns empty with no project set', async () => {
+    sinon.stub(ctx.project, 'projectId').resolves(undefined)
+
+    const result = await dataSource.getRelevantRuns([TEST_SHA])
+
+    expect(result).to.equal(EMPTY_RETURN)
+  })
+
+  //TODO: Skipping to figure out how to mock cloud query
```

Comment:
Could we track this with an issue number, or is this still a TODO before this PR is merged?

Reply:
Tests added